Automattic/vip-go-mu-plugins
# Adding Parse.ly telemetry

## Description

This code was initially written by @jblz. 

We are introducing telemetry into the Parse.ly plugin to better understand how the plugin is used and how it can be improved. This functionality is opt-in via a filter (it is disabled by default).

This PR adds a telemetry framework that connects with Automattic's Tracks system. It is prepared to be extended with other telemetry back-ends. It currently tracks a small set of events happening in `wp-admin` with respect to the Parse.ly plugin, but we expect this events to grow in the future.

## Changelog Description

### Adding Parse.ly telemetry

Setting up telemetry infrastructure for Parse.ly using A8c tracks.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Check out PR.
2. Enable Parse.ly and Tracking by:

```
add_filter( 'wpvip_parsely_load_mu', '__return_true' );
add_filter( 'wp_parsely_enable_telemetry_backend', '__return_true' );
```

3. Open the Parse.ly settings panel in `wp-admin` and update some options. Check that the actions get logged in Tracks.
